sos/plugins/dnf.py

```diff
@@ -57,11 +57,11 @@ def setup(self):
             "package-cleanup --problems"
         ])
 
-        if self.get_option("history"):
+        if self.get_option("history") and not self.get_option("history-info"):
             self.add_cmd_output("dnf history")
 
         if self.get_option("history-info"):
-            history = self.call_ext_prog("dnf history")
+            history = self.get_cmd_output_now("dnf history")
             transactions = -1
             if history['output']:
```

Comment:
Although it is not directly relevant to the change in API, could we drop this test? IMHO iterating over (potentially empty) splitlines is sufficient..

Reply:
Fair point, will update